### ISSUE TITLE
Interpolation mangles up lists

### ISSUE THREAD

**Issue by SanderElias** [sass/libsass#639]
I had a strange bug, and tracked it down to LibSass.

`map-keys($primary-color-palette)`  result:
`'50', '100', '200', '300', '400', '500', '600', '700', '800', '900', 'A100', 'A200', 'A400', 'A700'`

`"#{map-values($primary-color-palette)}"` result:
`"5','10','20','30','40','50','60','70','80','90','A10','A20','A40','A700"`

[see it in Sassmeister](http://sassmeister.com/gist/b06e787e90df2558d1eb)

**Comment by lunelson** [comment]
As of today, I'm still seeing problems with lists of quoted items. The following is a bit elaborate but please bear with me, it's just to show that what's at issue is a list of explicitly quoted strings:
```scss
$contexts: (
  mobile-l: (min-width: 20em),
  tablet-s: (min-width: 30em),
  tablet-l: (min-width: 48em),
  desktop-s: (min-width: 64em),
  desktop-m: (min-width: 80em)
) !default;


$quoted-context-keys: ();
@each $key in map-keys($contexts) {
  $quoted-context-keys: append($quoted-context-keys, quote($key), 'comma');
}

.test {
  output: map-keys($contexts);
  output: $quoted-context-keys;
  output: #{$quoted-context-keys};
  output: "[#{$quoted-context-keys}]";
  output: "#{$quoted-context-keys}";
  output: "["#{$quoted-context-keys}"]";
  output: '#{$quoted-context-keys}';
  output: "['#{$quoted-context-keys}']";
}
```
in ruby sass 3.4.9 produces
```css
.test {
  output: mobile-l, tablet-s, tablet-l, desktop-s, desktop-m;
  output: "mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m";
  output: mobile-l, tablet-s, tablet-l, desktop-s, desktop-m;
  output: "[mobile-l, tablet-s, tablet-l, desktop-s, desktop-m]";
  output: "mobile-l, tablet-s, tablet-l, desktop-s, desktop-m";
  output: "["mobile-l, tablet-s, tablet-l, desktop-s, desktop-m"]";
  output: "mobile-l, tablet-s, tablet-l, desktop-s, desktop-m";
  output: "['mobile-l, tablet-s, tablet-l, desktop-s, desktop-m']"; }
```
but in sassc: 3.1.0-2-g2033 / libsass: 3.1.0-76-g71448, produces
```css
.test {
  output: mobile-l, tablet-s, tablet-l, desktop-s, desktop-m;
  output: "mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m";
  output: mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m;
  output: "[mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m]";
  output: "mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m";
  output: "["mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m"]";
  output: 'mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m';
  output: "['mobile-l", "tablet-s", "tablet-l", "desktop-s", "desktop-m']"; }
```
note different results with single and double quoting, etc.

**Comment by xzyfer** [comment]
Thanks for this @lunelson . This appears to be a different issue than the one originally reported.

Could you please create a new issue and spec?

**Comment by lunelson** [comment]
Sure thing

**Comment by mgreter** [comment]
This should be solved by #910!
